src/test/java/be/seeseemelk/mockbukkit/block/data/SwitchMockTest.java

```diff
@@ -0,0 +1,144 @@
+package be.seeseemelk.mockbukkit.block.data;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.Set;
+
+import org.bukkit.Material;
+import org.bukkit.Tag;
+import org.bukkit.block.BlockFace;
+import org.bukkit.block.data.FaceAttachable.AttachedFace;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import be.seeseemelk.mockbukkit.MockBukkit;
+
+class SwitchMockTest
+{
+
+	SwitchMock switchMock;
+
+	@BeforeEach
+	void setUp()
+	{
+		MockBukkit.mock();
+		switchMock = new SwitchMock(Material.ACACIA_BUTTON);
+	}
+
+	@AfterEach
+	void tearDown()
+	{
+		MockBukkit.unmock();
+	}
+
+	@Test
+	void constructor_DefaultValues()
+	{
+		assertEquals(BlockFace.NORTH, switchMock.getFacing());
+		assertFalse(switchMock.isPowered());
+	}
+
+	@Test
+	void constructor_Material()
+	{
+		for (Material button : Tag.BUTTONS.getValues())
+		{
+			assertDoesNotThrow(() -> new SwitchMock(button));
+		}
+		assertDoesNotThrow(() -> new SwitchMock(Material.LEVER));
+
+	}
```

Comment:
Maybe a Parameterized Test would be more appropriate for this, since it would allow individual cases to fail, making debugging easier

Reply:
[Parameterized Tests](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests)

Reply:
Did not know that existed, one question: How do I specify all values in a tag for a parameterized test? Like I found the EnumSource anotation, but that does not seem to support what I want to do.